ibm/service/power/resource_ibm_pi_volume_test.go

```diff
@@ -160,3 +160,64 @@ func testAccCheckIBMPIVolumePoolConfig(name string) string {
 	  }
 	`, name, acc.Pi_cloud_instance_id)
 }
+
+func TestAccIBMPIVolumeGRS(t *testing.T) {
+	name := fmt.Sprintf("tf-pi-volume-%d", acctest.RandIntRange(10, 100))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMPIVolumeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMPIVolumeGRSConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMPIVolumeExists("ibm_pi_volume.power_volume"),
+					resource.TestCheckResourceAttr(
+						"ibm_pi_volume.power_volume", "pi_volume_name", name),
+					resource.TestCheckResourceAttr(
+						"ibm_pi_volume.power_volume", "pi_replication_enabled", "true"),
+					resource.TestCheckResourceAttr(
+						"ibm_pi_volume.power_volume", "replication_type", "global"),
+				),
+			},
+			{
+				Config: testAccCheckIBMPIVolumeGRSUpdateConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMPIVolumeExists("ibm_pi_volume.power_volume"),
+					resource.TestCheckResourceAttr(
+						"ibm_pi_volume.power_volume", "pi_volume_name", name),
+					resource.TestCheckResourceAttr(
+						"ibm_pi_volume.power_volume", "pi_replication_enabled", "false"),
+					resource.TestCheckResourceAttr(
+						"ibm_pi_volume.power_volume", "replication_type", ""),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMPIVolumeGRSConfig(name string) string {
+	return fmt.Sprintf(`
+	resource "ibm_pi_volume" "power_volume"{
+		pi_volume_size       = 20
+		pi_volume_name       = "%[1]s"
+		pi_volume_pool       = "%[3]s"
+		pi_volume_shareable  = true
+		pi_cloud_instance_id = "%[2]s"
+		pi_replication_enabled = true
+	  }
+	`, name, acc.Pi_cloud_instance_id, acc.PiStoragePool)
+}
+
+func testAccCheckIBMPIVolumeGRSUpdateConfig(name string) string {
+	return fmt.Sprintf(`
+	resource "ibm_pi_volume" "power_volume"{
+		pi_volume_size       = 20
+		pi_volume_name       = "%[1]s"
+		pi_volume_pool       = "%[3]s"
+		pi_volume_shareable  = true
+		pi_cloud_instance_id = "%[2]s"
+		pi_replication_enabled = false
+	  }
+	`, name, acc.Pi_cloud_instance_id, acc.PiStoragePool)
+}
```

Comment:
These 2 funcs can be combined with`replication` as a parameter.

Reply:
test case config refactored